src/dune_rules/resolve.ml

```diff
@@ -41,6 +41,8 @@ let to_result x = x
 
 let of_error x = Error x
 
+let raise_error ({ exn; _ } : error) = raise exn
```

Comment:
Hmm, it's not right to drop the stack frames. They need to be included in the error somehow. Too bad we don't have any convenient api's to do the right thing.

There's the private `Report_error.format_memo_stacks` to turn these frames into something readable. I also don't think we need a general exception type for `Resolve.error`. Keeping it a `User_message.t` would allow us to easily append some more readable information.

We don't need to solve all of these issues right away I suppose.

Reply:
I pushed a 3rd commit that takes care of this. Let me know what you think. It's a bit more invasive than I'd like, and I chose to duplicate the stack frame printer function, but it seems to work well.

Reply:
Looks right, but it seems to bork the formatting of some errors. See tests.

Reply:
On second thought, there's an easy way to solve this problem:

1. introduce `push_frames : error -> unit Memo.t`
2. call this `Resolve.push_frames` before creating the rule for the dummy target that will fail.

I still like your refactoring in Resolve.t though. But that can go in a separate PR.

Reply:
Makes sense. Pushed the version with `push_frames`, will move the refactoring to a separate PR.